### PR TITLE
fix dn parsing without attr or value

### LINKED ISSUE #114: Problem with ParseDN
I'm using ParseDN to validate the DN format. from my understanding once ParseDN code hits ',' or '+' chars, the code assigns attribute.Value from the buffer and appends attribute to RDN.Attributes. This doesn't check whether attribute.Type is valid or not nil before appending.

The code  @ https://github.com/go-ldap/ldap/blob/v2.5.0/dn.go#L147. If I enter "test,DC=example,DC=com" as DN it doesn't raise any errors even though its invalid DN input.

### PR DESCRIPTION
**Pull request by vetinari**
fixes #114

### CODE REVIEW

**Comment by liggitt** [comment]
LGTM, squash before merge

**Comment by vetinari** [comment]
@liggitt done

File: dn.go [hunk 1]
```diff
@@ -143,6 +143,9 @@ func ParseDN(str string) (*DN, error) {
 			}
 		} else if char == ',' || char == '+' {
 			// We're done with this RDN or value, push it
+			if len(attribute.Type) == 0 {
+				return nil, errors.New("incomplete type, value pair")
```
Comment (liggitt) [inline dn.go]: just had a thought... should we include `stringFromBuffer()` in the error to aid debugging the bad attribute? or if we don't want to allow injecting arbitrary values into the message, at least include the position?

Comment (liggitt) [inline dn.go]: I guess we didn't include the bad attribute in the "DN ended with incomplete type, value pair" message

Comment (liggitt) [inline dn.go]: do we also need to check for `len(attribute.Type) == 0` in the `if char == '='` block, to catch `=foo,DC=example,DC=com`?

Comment (vetinari) [inline dn.go]: That's caught here or with the "DN ended with incomplete type, value pair" error, adding 
`"=test,DC=example,DC=com": "incomplete type, value pair",` in `dn_test.go:TestErrorDNParsing()` passes.

Comment (vetinari) [inline dn.go]: added as test and squashed

**Comment by liggitt** [comment]
thanks, looks like it needs a rebase on top of your other PR

**Comment by vetinari** [comment]
done...

**Pull request merged**